### PR TITLE
Warning and styling clean up Vue 3

### PR DESCRIPTION
**Pull request by M-Kusumgar**
## Motivation
Alright, hopefully this one won't be that bad. There may be a lot of files changed but nothing crazy, just some styling issues, warning fixes, etc. It is called Demo Vue 3 because I used this branch for the demo to jeff and rachel.

## Guide
This time I will just add comment in the files where I think it need explanations. Just a bunch of smaller styling issues, fixing prop types so vue doesn't throw warnings (when we pass null into a prop with type String for example), and other warnings in console

## Checklist

- [ ] I have incremented version number, or version needs no increment
- [ ] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/package.json [hunk 1]
```diff
@@ -105,12 +105,15 @@
     "webpack-node-externals": "^3.0.0"
   },
   "dependencies": {
+    "@m-kusumgar/vue3-treeselect": "^1.0.18",
```
Comment (M-Kusumgar) [inline src/app/static/package.json:108]: We can change this to reside's npm account, I just didn't have the details so I just made my own npm account. The treeselect component was throwing a couple of warnings like $createElement was unused or Transition tag not recognised, little things in the source code so I changed the source code. Note, I have not changed the functionality, just removed unused variables from the output or replaced a couple transition tags with divs, that is all I did to remove the warnings

File: src/app/static/src/app/components/adr/SelectRelease.vue [hunk 2]
```diff
@@ -7,9 +7,9 @@
                 value="useLatest"
                 v-model="choiceADR"
             />
-            <label for="useLatest" v-translate="'useLatest'" class="pr-1"></label>
+            <label for="useLatest" v-translate="'useLatest'" style="margin-left: 4px;"></label>
```
Comment (M-Kusumgar) [inline src/app/static/src/app/components/adr/SelectRelease.vue:10]: padding right class wasnt working in a couple of places, we do use other bootstrap classes so wasnt sure why but just replaced the styles with some small pixel values, I think it looks just fine

File: src/app/static/src/app/components/genericChart/GenericChart.vue [hunk 3]
```diff
@@ -60,12 +60,12 @@
...
-                                        :unfiltered-data="unfilteredData ? unfilteredData[dataSource.config.id] : []"></download-indicator>
-                    <generic-chart-table :table-config="dataSource.tableConfig || { columns: [] }"
+                    <download-indicator :filtered-data="filteredDataWithoutPages![dataSource.config.id]"
+                                        :unfiltered-data="unfilteredData[dataSource.config.id]"></download-indicator>
+                    <generic-chart-table :table-config="dataSource.tableConfig!"
                                          :filtered-data="chartData[dataSource.config.id]"
-                                         :columns="dataSource.columns || []"
-                                         :selected-filter-options="dataSource.selections.selectedFilterOptions || {}"
+                                         :columns="dataSource.columns!"
+                                         :selected-filter-options="dataSource.selections.selectedFilterOptions!"
```
Comment (M-Kusumgar) [inline src/app/static/src/app/components/genericChart/GenericChart.vue:68]: just reverting this section back to how it actually was in hint dw too much about this

File: src/app/static/src/app/components/genericChart/Plotly.vue [hunk 4]
```diff
@@ -87,27 +87,26 @@ import { StyleValue } from "vue";
...
-                        this.rendering = false;
-                    }
-                }, 0);
+                const el = this.$refs.chart;
+                const drawFunc = Plotly.react;
+                this.layoutRequired = false;
+                const dataScatterGL = this.data.data.map((data: any) => {return {...data, type: "scattergl"}})
+                console.log(dataScatterGL)
+                await drawFunc(el as HTMLElement, dataScatterGL, this.data.layout, {...this.data.config as any});
+                this.rendering = false;
```
Comment (M-Kusumgar) [inline src/app/static/src/app/components/genericChart/Plotly.vue:98]: this was my initial attempt at fixing the plotly graph, ignore this, this gets a PR that will be up shortly, all these changes are going to be discarded

File: src/app/static/src/app/components/header/LanguageMenu.vue [hunk 5]
```diff
@@ -1,14 +1,14 @@
 <template>
     <drop-down :text="currentLanguage" :right="true" style="flex: none">
-        <a class="dropdown-item" href="#" v-on:mousedown="() => changeLanguage('en')">
+        <p class="dropdown-item" tabindex="0" v-on:mousedown="() => changeLanguage('en')" style="margin: 0;">
```
Comment (M-Kusumgar) [inline src/app/static/src/app/components/header/LanguageMenu.vue:3]: same bug as #835 causing user to go back to the homepage

File: src/app/static/src/app/components/plots/FilterSelect.vue [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
+            <label :class="['font-weight-bold', { 'disabled-label': disabled }]" v-translate="label"></label>
+            <span v-if="labelTooltip"
+                v-tooltip="{
+                        content: `<dl>${labelTooltip}</dl>`,
+                        // Keep the tooltip open when mouse over the popup
+                        popperTriggers: ['hover'],
+                    }">
+                <vue-feather type="help-circle" size="20" style="margin-left: 4px;"></vue-feather>
+            </span>
+        </div>
```
Comment (M-Kusumgar) [inline src/app/static/src/app/components/plots/FilterSelect.vue:13]: centering stuff

File: src/app/static/src/app/components/plots/ResetMap.vue [hunk 7]
```diff
@@ -6,7 +6,9 @@
                 href="#" :title="tooltipContent('resetView')" 
                 role="button" 
                 :aria-label="tooltipContent('resetView')">
-                <vue-feather type="refresh-cw" size="20"></vue-feather>
+                <vue-feather type="refresh-cw"
+                             size="20"
+                             style="vertical-align: middle; margin-bottom: 3px; margin-right: 1px;"></vue-feather>
```
Comment (M-Kusumgar) [inline src/app/static/src/app/components/plots/ResetMap.vue:11]: even centering the reset map icon normally is slightly off, might be to do with vue feather so need a manual amount of centering

File: src/app/static/src/app/components/projects/ProjectHistory.vue [hunk 8]
```diff
@@ -123,27 +123,27 @@
                     <div class="col-md-2 version-cell version-updated-cell">
                         {{ format(v.updated) }}
                     </div>
-                    <div class="col-md-1 version-cell load-cell"
-                         v-tooltip="getTranslatedValue('load')">
+                    <div class="col-md-1 version-cell load-cell">
                         <button class=" btn btn-sm btn-red-icons"
+                                v-tooltip="getTranslatedValue('load')"
```
Comment (M-Kusumgar) [inline src/app/static/src/app/components/projects/ProjectHistory.vue:128]: tooltips were misaligned because they were on the div not the button!

File: src/app/static/src/app/components/projects/ProjectHistory.vue [hunk 9]
```diff
@@ -20,13 +20,13 @@
                         id="version-toggle"
                         :aria-label="`toggle ${getTranslatedValue('versionCountLabelSingle')} ${p.id}`"
                         class="btn btn-xs bg-transparent shadow-none py-0"
-                        @click="toggleVersionMenu">
-                        <vue-feather v-show="!openVersion"
+                        @click="toggleVersionMenu(p.id)">
+                        <vue-feather v-show="!openVersion[p.id]"
```
Comment (M-Kusumgar) [inline src/app/static/src/app/components/projects/ProjectHistory.vue:24]: manually had to do a toggle because v-b-toggle was not working

File: src/app/static/src/app/store/translations/locales.ts [hunk 10]
```diff
@@ -533,7 +533,7 @@ const en: Translations = {
     noteHeader: "Notes: (your reason for saving as a new version)",
     notUsed: "Not used",
     ok: "OK",
-    or: "or",
+    or: " or",
```
Comment (M-Kusumgar) [inline src/app/static/src/app/store/translations/locales.ts:536]: makes sure padding between create new project or return to current project on projects page

File: src/app/static/package.json (hunk 1 above)
Comment (LekanAnni) [reply src/app/static/package.json:108]: Could this be changed in this PR?

File: src/app/static/src/app/components/adr/ADRIntegration.vue [hunk 11]
```diff
@@ -5,7 +5,7 @@
             <select-dataset></select-dataset>
             <div class="pt-3" id="adr-capacity" v-if="selectedDataset">
                 <span class="font-weight-bold align-self-stretch" v-translate="'adrAccessLevel'"></span>
-                <span v-tooltip="handleUploadPermission(hasUploadPermission, true)">
+                <span v-tooltip="handleUploadPermission(hasUploadPermission, true)" style="margin-left: 4px;">
```
Comment (LekanAnni) [inline src/app/static/src/app/components/adr/ADRIntegration.vue:8]: can you use bootstrap spacing such as `ml-4` instead?

File: src/app/static/src/app/components/adr/SelectDataset.vue [hunk 12]
```diff
@@ -1,9 +1,10 @@
 <template>
     <div class="d-flex">
-        <div v-if="selectedDataset" style="margin-top: 8px">
+        <div v-if="selectedDataset" style="margin-top: 8px;">
```
Comment (LekanAnni) [inline src/app/static/src/app/components/adr/SelectDataset.vue:3]: same

File: src/app/static/src/app/components/adr/SelectDataset.vue [hunk 13]
```diff
@@ -1,9 +1,10 @@
...
             <span class="font-weight-bold"
                   v-translate="'selectedDataset'"
-                  id="selectedDatasetSpan">
+                  id="selectedDatasetSpan"
+                  style="margin-right: 4px;">
```
Comment (LekanAnni) [inline src/app/static/src/app/components/adr/SelectDataset.vue:7]: here too

File: src/app/static/src/app/components/adr/SelectRelease.vue (hunk 2 above)
Comment (LekanAnni) [inline src/app/static/src/app/components/adr/SelectRelease.vue:10]: bootstraps spacing here too

File: src/app/static/src/app/components/downloadIndicator/DownloadIndicator.vue [hunk 14]
```diff
@@ -23,7 +23,7 @@
     }
 
     interface Props {
-        filteredData: unknown[] | null
+        filteredData?: unknown[] | null
```
Comment (LekanAnni) [inline src/app/static/src/app/components/downloadIndicator/DownloadIndicator.vue:26]:
```suggestion
        filteredData: unknown[] | null
```

There's no point making it optional from here when you already set required to false.

File: src/app/static/src/app/components/projects/ProjectHistory.vue [hunk 15]
```diff
@@ -297,7 +297,7 @@
         renameProjectNote: string;
         projectNoteToEdit: number | null;
         editedProjectNote: string;
-        openVersion: boolean
+        openVersion: Record<number, boolean>
```
Comment (LekanAnni) [inline src/app/static/src/app/components/projects/ProjectHistory.vue:300]: Why did you change this ?

**Review by LekanAnni — CHANGES_REQUESTED** [review]
Looking good,

You have added a lot of styles, could you replace styles with bootstraps spacing instead? also the optional ? make on props can be removed when required is set to false.

File: src/app/static/src/app/components/adr/ADRIntegration.vue (hunk 11 above)
Comment (M-Kusumgar) [reply src/app/static/src/app/components/adr/ADRIntegration.vue:8]: done

File: src/app/static/src/app/components/adr/SelectDataset.vue (hunk 12 above)
Comment (M-Kusumgar) [reply src/app/static/src/app/components/adr/SelectDataset.vue:3]: done

File: src/app/static/src/app/components/adr/SelectDataset.vue (hunk 13 above)
Comment (M-Kusumgar) [reply src/app/static/src/app/components/adr/SelectDataset.vue:7]: done

File: src/app/static/src/app/components/adr/SelectRelease.vue (hunk 2 above)
Comment (M-Kusumgar) [reply src/app/static/src/app/components/adr/SelectRelease.vue:10]: done

File: src/app/static/src/app/components/downloadIndicator/DownloadIndicator.vue (hunk 14 above)
Comment (M-Kusumgar) [reply src/app/static/src/app/components/downloadIndicator/DownloadIndicator.vue:26]: ahhh I have to, so the defineComponentVue2 uses the fact that it is optional prop and enforces that required is false, dont worry next PR we will completely remove all of this business with the defineComponentVue2 so you can safely ignore this one!

File: src/app/static/src/app/components/projects/ProjectHistory.vue (hunk 15 above)
Comment (M-Kusumgar) [reply src/app/static/src/app/components/projects/ProjectHistory.vue:300]: so i realised when you have multiple projects, they sahre this openVersion prop hence when you expand one projects versions, they all expand, so i need to change it to an object of booleans that store whether each of them are open or not

**Comment by M-Kusumgar** [comment]
> You have added a lot of styles, could you replace styles with bootstraps spacing instead?

So if I just import bootstrap css or scss then there are styles clashing, not sure why but things have the wrong colors. I have imported `bootstrap-grid.scss` which has the spacings but does not interfere with the colors. Bootstrap uses `rem` units for their spacing and generally multiples of 4 pixels can equate to one `rem` so I have replaced the margins wherever they are in multiples of 4

File: src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlSection.vue [hunk 16]
```diff
@@ -2,9 +2,9 @@
     <div>
         <h3 @click="toggleSection" :class="{'cursor-pointer': controlSection.collapsible}">
             {{controlSection.label}}
-            <component v-if="controlSection.collapsible"
+            <vue-feather v-if="controlSection.collapsible"
```
Comment (LekanAnni) [inline src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlSection.vue:5]: Needs testing

File: src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlSection.vue [hunk 17]
```diff
@@ -113,15 +83,15 @@
         computed: {
             chevronComponent() {
                 if (this.open) {
-                    return "chevron-up-icon"
+                    return "chevron-up"
                 }
-                return "chevron-down-icon"
+                return "chevron-down"
```
Comment (LekanAnni) [inline src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlSection.vue:88]: Missing test ?

File: src/app/static/src/app/vue-dynamic-form/src/DynamicFormSelect.vue [hunk 18]
```diff
@@ -1,21 +1,17 @@
...
-                :value="opt.id">
-            {{opt.label}}
-        </option>
-    </b-form-select>
+    <hint-tree-select
+            :model-value="value"
+            :multiple="false"
+            :clearable="false"
+            @update:model-value="newVal => value = newVal"
+            :options="formOptions">
```
Comment (LekanAnni) [inline src/app/static/src/app/vue-dynamic-form/src/DynamicFormSelect.vue:7]: Can you add a test asserting props ?

**Review by LekanAnni — CHANGES_REQUESTED** [review]
Thanks for adding some changes,

Sorry I didn't make a comment on these earlier, I have identified few others.

File: src/app/static/src/app/components/files/ManageFile.vue [hunk 19]
```diff
@@ -2,7 +2,8 @@
     <div class="form-group">
         <label class="font-weight-bold mb-0" v-translate="label"></label>
         <span id="required" v-if="required"
-              class="small"
+              class="ml-1"
+              style="font-size: small;"
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/files/ManageFile.vue:6]: unfortunately class small doesnt work, must not be included in bootstrap-grid.scss

File: src/app/static/src/app/components/genericChart/GenericChart.vue [hunk 20]
```diff
@@ -60,12 +60,12 @@
                 </div>
                 <div v-for="dataSource in chartConfigValues.dataSourceConfigValues.filter(ds => ds.tableConfig)"
                      :key="dataSource.config.id">
-                    <download-indicator :filtered-data="filteredDataWithoutPages ? filteredDataWithoutPages[dataSource.config.id] : null"
-                                        :unfiltered-data="unfilteredData ? unfilteredData[dataSource.config.id] : []"></download-indicator>
-                    <generic-chart-table :table-config="dataSource.tableConfig || { columns: [] }"
+                    <download-indicator :filtered-data="filteredDataWithoutPages![dataSource.config.id]"
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/genericChart/GenericChart.vue:63]: so in master there is no non null assertion but this may cause a run time vue props warning, I can remove these but console might then contain some more warnings

File: src/app/static/src/app/components/genericChart/Plotly.vue [hunk 21]
```diff
@@ -87,27 +87,24 @@ import { StyleValue } from "vue";
             }
         },
         methods: {
-            drawChart: function() {
+            drawChart: async function() {
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/genericChart/Plotly.vue:90]: ignore this whole file, #859 will change the whole file and there will be tests for it in that PR, no point doing tests for this here

File: src/app/static/src/app/components/plots/ResetMap.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/plots/ResetMap.vue:11]: I cannot use bootstrap here, it doesn't work with pixels, i need finer control over this one in particular as it is just tiny adjustments to have the reset map icon completely in center of the square on maps

[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/plots/ResetMap.vue:11]: see my comment earlier about bootstrap spacing units and multiples of 4

File: src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlGroup.vue [hunk 22]
```diff
@@ -2,9 +2,11 @@
     <b-row class="my-2">
         <label v-if="controlGroup.label" class="col-form-label col-md-5">{{controlGroup.label}}
             <span v-if="helpText" class="icon-small" v-tooltip="helpText">
-                    <vue-feather type="help-circle"></vue-feather>
+                    <vue-feather type="help-circle" style="vertical-align: middle;"></vue-feather>
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlGroup.vue:5]: ignore vue-dynamic-form and vue-charts for testing as these will eventually be from packages, which will have their own testing

File: src/app/static/src/app/components/modelRun/ModelRun.vue [hunk 23]
```diff
@@ -5,9 +5,10 @@
                 :disabled="running"
                 v-translate="'fitModel'">
         </button>
-        <h4 v-if="complete" class="mt-3" id="model-run-complete" v-translate="'fittingComplete'">
+        <div v-if="complete" class="mt-3 d-flex align-items-center" id="model-run-complete">
+            <h4 v-translate="'fittingComplete'" style="margin-right: 7px;"></h4>
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/modelRun/ModelRun.vue:9]: replaced styling, this is another example of code coverage being wonky i think, the only thing that wasnt tested is the classes for the div, I have added that one line of test but still complaining that this line isn't covered

File: src/app/static/src/app/components/plots/choropleth/Choropleth.vue [hunk 24]
```diff
@@ -3,13 +3,13 @@
         <filters class="col-md-3" v-if="includeFilters"
                  :filters="filters"
                  :selectedFilterOptions="selections.selectedFilterOptions"
-                 @update="onFilterSelectionsChange"></filters>
+                 @update:filters="onFilterSelectionsChange"></filters>
         <div id="chart" :class="includeFilters ? 'col-md-9' : 'col-md-12'">
-            <l-map ref="map" style="height: 800px; width: 100%" @ready="updateBounds" @vnode-updated="updateBounds">
+            <l-map ref="map" style="height: 800px; width: 100%" @ready="updateBounds">
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/plots/choropleth/Choropleth.vue:8]: I have checked the styling in the map, we use a mock import for the map in the test so cannot check the @ready bit of the line but we shouldn't have to as long as we test updateBounds since we do not trigger the @ready event

File: src/app/static/src/app/components/plots/choropleth/Choropleth.vue [hunk 25]
```diff
@@ -429,6 +429,9 @@
             },
         beforeMount() {
             this.initialise();
-        }
+        },
+        updated() {
+            this.updateBounds();
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/plots/choropleth/Choropleth.vue:434]: i have included a test for this but we really shouldn't need to test if a vue lifecycle hook executes the code block inside it, as long as we test updateBounds, we should not need to test the framework/library

File: src/app/static/src/app/components/projects/ShareProject.vue [hunk 26]
```diff
@@ -34,23 +33,29 @@
...
+                            @click="confirmShareProject"
+                            :disabled="invalidEmails || cloningProject"
+                            v-translate="'ok'">
+                    </button>
+                    <button type="button"
+                            class="btn btn-white"
+                            @mousedown="cancelShareProject"
+                            :disabled="cloningProject"
+                            v-translate="'cancel'"
+                            style="margin-left: 10px;">
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/projects/ShareProject.vue:55]: done

File: src/app/static/src/app/vue-dynamic-form/src/DynamicFormControl.vue [hunk 27]
```diff
@@ -4,9 +4,11 @@
             <span v-if="formControl.helpText"
                   class="icon-small"
                   v-tooltip="formControl.helpText">
-                <vue-feather type="help-circle"></vue-feather>
+                <vue-feather type="help-circle" style="vertical-align: middle;"></vue-feather>
             </span>
-            <span v-if="formControl.required" class="small" :class="{'text-danger': valueIsEmpty(formControl.value)}">({{requiredText}})</span>
+            <span v-if="formControl.required"
+                  :class="{'text-danger': valueIsEmpty(formControl.value)}"
+                  style="font-size: small; margin-left: 4px;">({{requiredText}})</span>
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/vue-dynamic-form/src/DynamicFormControl.vue:11]: done

File: src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlGroup.vue [hunk 28]
```diff
@@ -2,9 +2,11 @@
...
                 </span>
-            <span v-if="required" class="small" :class="{'text-danger': anyValueEmpty(controlGroup)}">({{requiredText}})</span>
+            <span v-if="required"
+                  :class="{'text-danger': anyValueEmpty(controlGroup)}"
+                  style="margin-left: 4px; font-size: small;">({{requiredText}})</span>
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlGroup.vue:9]: done

File: src/app/static/src/app/components/projects/ProjectHistory.vue (hunk 9 above)
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/projects/ProjectHistory.vue:24]: done

File: src/app/static/package.json (hunk 1 above)
Comment (EmmaLRussell) [reply src/app/static/package.json:108]: @M-Kusumgar If you send me your npm user name I'll add you to the reside-ic organisation. Is it just `m-kusumgar`?

Comment (M-Kusumgar) [reply src/app/static/package.json:108]: Yep I believe it is just m-kusumgar

File: src/app/static/src/app/components/Tick.vue [hunk 29]
```diff
@@ -1,4 +1,5 @@
 <template>
+    <!-- stop inheritance of attributes -->
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/Tick.vue:2]: sorry I have removed that, there is an error in console about svg having width and height so I thought it was automatically inheriting attributes put on Tick.vue (which would happen if there is only one parent component instead template and a comment would stop that from happening). However this didn't work anyway, so I have removed the comment

File: src/app/static/src/app/components/adr/SelectRelease.vue [hunk 30]
```diff
@@ -7,9 +7,9 @@
                 value="useLatest"
                 v-model="choiceADR"
             />
-            <label for="useLatest" v-translate="'useLatest'" class="pr-1"></label>
-            <span class="icon-small" v-tooltip="translate('datasetTooltip')">
-                <vue-feather type="help-circle"></vue-feather>
+            <label for="useLatest" v-translate="'useLatest'" class="ml-1"></label>
+            <span class="icon-small ml-1" v-tooltip="translate('datasetTooltip')">
+                <vue-feather type="help-circle" style="vertical-align: middle"></vue-feather>
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/adr/SelectRelease.vue:12]: this does work! will change this in the relevant places

File: src/app/static/src/app/components/adr/SelectRelease.vue [hunk 31]
```diff
@@ -20,13 +20,9 @@
...
-            <label
-                for="useRelease"
-                v-translate="'useRelease'"
-                class="pr-1"
-            ></label>
-            <span class="icon-small" v-tooltip="translate('releaseTooltip')">
-                <vue-feather type="help-circle"></vue-feather>
+            <label for="useRelease" v-translate="'useRelease'" class="ml-1"></label>
+            <span class="icon-small ml-1" v-tooltip="translate('releaseTooltip')">
+                <vue-feather type="help-circle" style="vertical-align: middle;"></vue-feather>
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/adr/SelectRelease.vue:25]: done

File: src/app/static/src/app/components/projects/ShareProject.vue [hunk 32]
```diff
@@ -34,23 +33,28 @@
                 <loading-spinner size="sm"></loading-spinner>
             </div>
             <template v-slot:footer>
+            <div style="display: flex; align-items: center; justify-content: space-between;">
                 <div class="text-muted help-text"
                      v-show="showValidationMessage"
-                     v-translate="'emailMultiValidation'">
+                     v-translate="'emailMultiValidation'"
+                     style="float: left; width: 60%;">
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/projects/ShareProject.vue:40]: done, btw sorry about all the inline styles, I am familiar with bootstrap but importing the whole bootstrap scss causes the colors of the site to be weird, no clue why!

File: src/app/static/src/app/components/projects/ShareProject.vue [hunk 33]
```diff
@@ -34,23 +33,28 @@
...
-                        :disabled="invalidEmails || cloningProject"
-                        v-translate="'ok'">
-                </button>
-                <button type="button"
-                        class="btn btn-white"
-                        @mousedown="cancelShareProject"
-                        :disabled="cloningProject"
-                        v-translate="'cancel'">
-                </button>
+                <div style="float: right;">
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/projects/ShareProject.vue:43]: done

File: src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlSection.vue (hunk 16 above)
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/vue-dynamic-form/src/DynamicFormControlSection.vue:5]: what are you uncertain about? would be useful to know as I will keep these in mind when I work on modifying the vue 3 version of vue-dynamic-form

File: src/app/static/src/app/components/plots/bubble/BubblePlot.vue [hunk 34]
```diff
@@ -42,8 +42,7 @@
             <l-map ref="map" style="height: 800px; width: 100%" @ready="updateBounds" @vnode-updated="updateBounds">
                 <template v-for="feature in currentFeatures" :key="feature.id">
                     <l-geo-json :geojson="feature"
-                                :options="options"
-                                :optionsStyle="style">
+                                :optionsStyle="() => {return style}">
```
[the remark this answers is not in the document]
Comment (M-Kusumgar) [reply src/app/static/src/app/components/plots/bubble/BubblePlot.vue:45]: yep does work!

**Comment by M-Kusumgar** [comment]
> The ADR access keys buttons are squashed together but they're going away with SSO so I guess that doesn't matter.

Yep this is why I havent really bothered with it!

> No error bars on the barcharts but that's going to be addressed in a future branch?

Hehe yes, #860 dw, coming up shortly!

> The icon in the File menu -> Load Model Outputs is a bit squished next to the text... oh, and actually nothing happens when I click that!

Yes! Just fixed it!

**Review by LekanAnni — APPROVED** [review]
LGTM 👌 

Thanks

File: src/app/static/src/app/components/genericChart/GenericChart.vue [hunk 35]
```diff
@@ -60,7 +60,7 @@
                 </div>
                 <div v-for="dataSource in chartConfigValues.dataSourceConfigValues.filter(ds => ds.tableConfig)"
                      :key="dataSource.config.id">
-                    <download-indicator :filtered-data="filteredDataWithoutPages![dataSource.config.id]"
+                    <download-indicator :filtered-data="filteredDataWithoutPages ? filteredDataWithoutPages[dataSource.config.id] : []"
```
Comment (EmmaLRussell) [inline src/app/static/src/app/components/genericChart/GenericChart.vue:63]: I thought it was safe to assume `filteredDataWithoutPages` would be truthy, so you could remove nullable option from the Computed interface?

**Review by EmmaLRussell — APPROVED** [review]
Looks good! Just one outstanding small comment, but fine to merge.

File: src/app/static/src/app/components/genericChart/GenericChart.vue (hunk 35 above)
Comment (M-Kusumgar) [reply src/app/static/src/app/components/genericChart/GenericChart.vue:63]: unfortunately the computed value actually explicitly returns a null in one of the cases, I tried replacing this with a `{}` but this caused errors, so I just put a ternary

**Comment by M-Kusumgar** [comment]
will merge but will also replace the package name in a packages PR, will be 2-3 PRs down the line
